src/services/ThreadPool.hxx

```diff
@@ -0,0 +1,98 @@
+#ifndef THREAD_POOL_HXX
+#define THREAD_POOL_HXX
+
+#include <vector>
+#include <queue>
+#include <memory>
+#include <thread>
+#include <mutex>
+#include <condition_variable>
+#include <future>
+#include <functional>
+#include <stdexcept>
+
+using Thread = std::thread;
+
+class ThreadPool {
+public:
+  explicit ThreadPool(uint32_t threads = 4) : initial_threads_num(threads) {}
+
+  inline void initialize(size_t threads = 4)
+  {
+    threads = std::max(initial_threads_num, threads);
```

Comment:
hm.... seems not, it an internal thread pool with workers
usually it created loke pool(4) for game tasks. pool(1) for loading tasks, pool(1) for sound task and other
if we create pool(cpu * 2) 90% it will kill performance and shouldnot work

Reply:
I think to have dedicated thread as we already have for sound is fine.

I was imagining using of thread pool as the possibility to parallelize the same task in multiple threads. For some calculation maybe. Like these Managers Liz created for building spawning and power.